src/client/feature.rs

```diff
@@ -43,13 +43,13 @@ impl Feature {
                 Value::Numeric(NumericValue(model_value.0.clone()))
             }
             crate::models::ValueKind::Boolean => {
-                Value::Boolean(model_value.0.as_bool().ok_or(Error::ProtocolError)?)
+                Value::Boolean(model_value.0.as_bool().ok_or(Error::ProtocolError("Expected Boolean".into()))?)
```

Comment:
If we want to, later we can add more typed information (`ErrorBooleanExpected`, `ErrorStringExpected`) so the message is just for humans... and we can also include what was the value we were trying to parse as boolean.